### PR TITLE
react structure setup

### PR DESCRIPTION
**Pull request by zuspan**

### CODE REVIEW

File: sub/app/.serverless/cloudformation-template-create-stack.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (richburdon) [inline sub/app/.serverless/cloudformation-template-create-stack.json]: Don't checkin .serverless to repo; add to .gitignore

File: sub/app/src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright 2018 FSM, Inc.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+
+class App extends React.PureComponent {
```
Comment (richburdon) [inline sub/app/src/components/App.js]: /** 
 * Main App
 */

I like to have comment blocks in front of all classes.

Also single new lines (not double).

File: sub/app/src/components/MassingTool/MassingTool.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Copyright 2018 FSM, Inc.
+ */
+
+
```
Comment (richburdon) [inline sub/app/src/components/MassingTool/MassingTool.css]: Remove newline.

File: sub/app/src/components/MassingTool/MassingTool.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+@import '../variables.css';
+
+.root {
+	background-color: red;
```
Comment (richburdon) [inline sub/app/src/components/MassingTool/MassingTool.css]: 2 space indent.

File: sub/app/src/components/MassingTool/MassingTool.css [hunk 5]
```diff
@@ -0,0 +1,13 @@
+/**
```
Comment (richburdon) [inline sub/app/src/components/MassingTool/MassingTool.css]: Use less (not css). Webpack is set-up for this.

File: sub/app/src/components/MassingTool/MassingTool.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import s from './MassingTool.css';
+
+//external components
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import Paper from 'material-ui/Paper';
+
+//components
+import View3D from '../View3D';
+
+class MassingTool extends React.Component {
```
Comment (richburdon) [inline sub/app/src/components/MassingTool/MassingTool.js]: 2 space indent (same comment about comment blocks and newlines).

File: sub/app/src/components/MassingTool/MassingTool.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+// Copyright 2018 FSM, Inc.
+//
+
+
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import s from './MassingTool.css';
+
+//external components
```
Comment (richburdon) [inline sub/app/src/components/MassingTool/MassingTool.js]: // External components. (space before).
// Single line comments should look like this.

File: sub/app/src/components/MassingTool/MassingTool.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+//components
+import View3D from '../View3D';
+
+class MassingTool extends React.Component {
+	// static PropTypes = {
+	// 	name: PropTypes.string.isRequired
+	// }
+
+	constructor(props) {
+		super(props);
```
Comment (richburdon) [inline sub/app/src/components/MassingTool/MassingTool.js]: super(...arguments) passes all arguments (and you don't have to declare them).
E.g., props is typically followed by context, etc.

File: sub/app/src/components/MassingTool/MassingTool.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+		this.state = {};
+	}
+
+	componentDidMount() {
```
Comment (richburdon) [inline sub/app/src/components/MassingTool/MassingTool.js]: add comment or TODO(zuspan): Fix something later, etc.

File: sub/app/src/components/View3D/View3D.css [hunk 10]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright 2018 FSM, Inc.
+ */
+
+
+@import '../variables.css';
+
+.root {
+	width: 100%;
```
Comment (richburdon) [inline sub/app/src/components/View3D/View3D.css]: I think it's better to use flex.
E.g., 

position: absolute;
top: 0; left: 0; right: 0; bottom: 0;
display: flex;

File: sub/app/src/components/View3D/package.json [hunk 11]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "View3D",
+  "version": "0.0.0",
+  "private": true,
+  "main": "./View3D.js"
```
Comment (richburdon) [inline sub/app/src/components/View3D/package.json]: filenames should be lower case and have underscores not camelCase.

E.g., view_3d.js

File: sub/app/src/main.js [hunk 12]
```diff
@@ -1,5 +1,48 @@
...
+  //   };
+  // },
+
+  // // Universal HTTP client
+  // fetch: createFetch(fetch, {
+  //   baseUrl: window.App.apiUrl,
+  // }),
+// };
+
+const container = document.getElementById('app-root');
```
Comment (richburdon) [inline sub/app/src/main.js]: the root ID was passed in the config from the server (which owns the template that defines the root div element).

File: sub/app/src/main.js [hunk 13]
```diff
@@ -1,5 +1,48 @@
...
+
+try{
+  
+	ReactDOM.render(
+		<App><MassingTool /></App>,
```
Comment (richburdon) [inline sub/app/src/main.js]: I prefer no space before `/>`

File: sub/app/src/main.js [hunk 14]
```diff
@@ -1,5 +1,48 @@
...
+		container,
+	)
+
+} catch(err){
+	if (__dev__){
```
Comment (richburdon) [inline sub/app/src/main.js]: space before brace.

Comment (zuspan) [reply sub/app/src/main.js]: meaning you like the space or prefer to remove the space?

File: sub/app/.serverless/cloudformation-template-create-stack.json (hunk 1 above)
Comment (zuspan) [reply sub/app/.serverless/cloudformation-template-create-stack.json]: done.

File: sub/app/src/components/App.js (hunk 2 above)
Comment (zuspan) [reply sub/app/src/components/App.js]: done for all components.

File: sub/app/src/components/MassingTool/MassingTool.css (hunk 3 above)
Comment (zuspan) [reply sub/app/src/components/MassingTool/MassingTool.css]: done

File: sub/app/src/components/MassingTool/MassingTool.css (hunk 4 above)
Comment (zuspan) [reply sub/app/src/components/MassingTool/MassingTool.css]: done

File: sub/app/src/components/MassingTool/MassingTool.js (hunk 7 above)
Comment (zuspan) [reply sub/app/src/components/MassingTool/MassingTool.js]: done

File: sub/app/src/components/MassingTool/MassingTool.js (hunk 9 above)
Comment (zuspan) [reply sub/app/src/components/MassingTool/MassingTool.js]: done

File: sub/app/src/components/View3D/View3D.css (hunk 10 above)
Comment (zuspan) [reply sub/app/src/components/View3D/View3D.css]: done

File: sub/app/src/components/View3D/package.json (hunk 11 above)
Comment (zuspan) [reply sub/app/src/components/View3D/package.json]: done

File: sub/app/src/main.js (hunk 12 above)
Comment (zuspan) [reply sub/app/src/main.js]: done.

File: sub/app/src/main.js (hunk 13 above)
Comment (zuspan) [reply sub/app/src/main.js]: done

File: sub/app/src/components/MassingTool/MassingTool.css (hunk 5 above)
Comment (zuspan) [reply sub/app/src/components/MassingTool/MassingTool.css]: I could not quickly figure out how to compile less with component style loading. Will pick this up next week, unless you know off the top of your head (I'm not a webpack master I'm afraid).

**Pull request merged**